addons/website_event/static/src/js/website_event.editor.js

```diff
@@ -2,11 +2,201 @@ odoo.define('website_event.editor', function (require) {
 "use strict";
 
 var core = require('web.core');
-var wUtils = require('website.utils');
+var time = require('web.time');
 var WebsiteNewMenu = require('website.newMenu');
-
+var Dialog = require('web.Dialog');
 var _t = core._t;
 
+var EventCreateDialog = Dialog.extend({
+    xmlDependencies: Dialog.prototype.xmlDependencies.concat(
+        [
+            '/website_event/static/src/xml/website_event_templates.xml'
+        ]
+    ),
+    template: 'website_event.create_event',
+    events: _.extend({}, Dialog.prototype.events, {
+        'change select[name="event_location"]': '_onLocationChanged',
+    }),
+
+    /**
+     * @override
+     * @param {Object} parent
+     * @param {Object} options
+     */
+    init: function (parent, options) {
+        options = _.defaults(options || {}, {
+            title: _t("New Event"),
+            size: 'medium',
+            buttons: [
+                {
+                    text: _t("Create"),
+                    classes: 'btn-primary',
+                    click: this.onCreateClick.bind(this),
+                },
+                {
+                    text: _t("Discard"),
+                    close: true
+                },
+            ]
+        });
+        this._super(parent, options);
+    },
+    start: function () {
+        var self = this;
+        return this._super.apply(this, arguments).then(function () {
+            self.$('div.o_event_form_datetime').each(function () {
+                self._initDateTimePicker($(this));
+            });
+
+            var $input = self.$('#address_id');
+            $input.select2({
+                width: '100%',
+                allowClear: true,
+                formatNoMatches: false,
+                multiple: false,
+                selection_data: false,
+                fill_data: function (query, data) {
+                    var that = this;
+                    var tags = {results: []};
+                    _.each(data, function (obj) {
+                        var display_text = obj.display_name + ': ' + obj.street + ', ' + (obj.zip ? (obj.zip + ', ') : '') + obj.city + ', ' + obj.country_id[1];
+                        if (that.matcher(query.term, display_text)) {
+                            tags.results.push({id: obj.id, text: display_text});
+                        }
+                    });
+                    query.callback(tags);
+                },
+                query: function (query) {
+                    var that = this;
+                    // fetch data only once and store it
+                    if (!this.selection_data) {
+                        self._rpc({
+                            model: 'res.partner',
+                            method: 'search_read',
```

Comment:
Can't you use display_name with 'address_inline' context key ? It normally adds the address in the name. I did not play with it since a while but it is worth a try I suppose (see _get_name in base/partner.py).

Reply:
Used the contact_address field instead which seems to do the job while being even simpler

Reply:
:+1: